api/src/main/java/org/openmrs/api/impl/PatientServiceImpl.java

```diff
@@ -574,10 +582,18 @@ public void mergePatients(Patient preferred, Patient notPreferred) throws APIExc
 			log.debug("Merge operation cancelled: Cannot merge user" + preferred.getPatientId() + " to self");
 			throw new APIException("Patient.merge.cancelled", new Object[] { preferred.getPatientId() });
 		}
+		
+		// do any "before-merge actions" that have been registered
+		if (patientMergeActions != null) {
+			for (PatientMergeAction patientMergeAction : patientMergeActions) {
+				patientMergeAction.beforeMergingPatients(preferred, notPreferred);
+			}
+		}
+		
 		requireNoActiveOrderOfSameType(preferred,notPreferred);
 		PersonMergeLogData mergedData = new PersonMergeLogData();
 		mergeVisits(preferred, notPreferred, mergedData);
-		mergeEncounters(preferred, notPreferred, mergedData);
+		mergeEncounters(preferred, notPreferred, mergedData);//merge the remaining encounter that don't have visit
```

Comment:
Was this comment part of the ticket?

Reply:
Not actually i just thought i should make it descriptive. But i will remove it while fixing the reviews Wluyima provided